File: celegans_delta.m
clear all;
close all;

load('celegans_delta.mat');
[X,Y] = meshgrid(d1,d2);

figure;
surf(X,Y,omg');
xlabel('delta 1');
ylabel('delta 2');
zlabel('objective');
shading interp;
view(-125,45);

figure;
surf(X,Y,doe');
xlabel('delta 1');
ylabel('delta 2');
zlabel('epsilon derivative');
shading interp;
view(-125,45);

figure;
surf(X,Y,epd_rmse');
xlabel('delta 1');
ylabel('delta 2');
zlabel('gradient rmse');
shading interp;
view(-125,45);

figure;
surf(X,Y,tog');
xlabel('delta 1');
ylabel('delta 2');
zlabel('objective time');
shading interp;
view(-125,45);

figure;
surf(X,Y,tgd');
xlabel('delta 1');
ylabel('delta 2');
zlabel('gradient time');
shading interp;
view(-125,45);

File: celegans_delta.jl
include("testStochastic.jl")
d1 = 1.00:0.10:5.00
d2 = 0.05:0.01:0.65
omg, doe, epd, tog, tgd, omg_nev, doe_nev, epd_nev = accuracy_celegans(d1,d2)
epd_rmse = [norm(epd[i,j,:]-epd_nev)/sqrt(length(epd_nev)) for i in 1:length(d1), j in 1:length(d2)]
MAT.matwrite("celegans_delta.mat", Dict("d1"=>collect(d1), "d2"=>collect(d2), "omg"=>omg, "doe"=>doe, "epd"=>epd, "epd_rmse"=>epd_rmse, "tog"=>tog, "tgd"=>tgd))
